### ISSUE TITLE
[MAINTENANCE] Update Crystal Version to Latest Stable (currently v1.9.2)

### ISSUE THREAD

**Issue by agentpoyo** [cncf/cnf-testsuite#1814]
**Describe the software update**
The last crystal update to v1.6.0 was back in December of 2022. The latest stable release now at issue creation is v1.9.2.

**Describe why update is needed**
Crystal should be updated to support the latest stable releases to not fall behind release cycles while older versions start to be deprecated on OS updates and releases.

**What version is in use now?**
1.6.0

**What version is the latest version and/or which version you will use for the update?**
1.9.2

**Additional context**
We should also update the latest shards modules in use by the cnf-testsuite where applicable.

**Comment by HashNuke** [comment]
I've submitted a PR to retriable.cr shard repo - https://github.com/Sija/retriable.cr/pull/12.

From what I checked, that seems to be the only shard that requires an update. Once the PR is merged, I'll submit another PR to update the crystal version and the ameba version in the testsuite.

**Comment by HashNuke** [comment]
Created the branch `maintain/1814` for changes to this ticket. Will send PR after #1822 is resolved because the main branch builds are failing.

**Comment by lixuna** [comment]
Hi @HashNuke and @agentpoyo - what is the status of updating Crystal?

If done, can the related PR for "Bump crystal-ameba/ameba to 1.5.0" be addressed next? Thanks!

**Comment by agentpoyo** [comment]
> I've submitted a PR to retriable.cr shard repo - [Sija/retriable.cr#12](https://github.com/Sija/retriable.cr/pull/12).

Looks like they merged the shard that @HashNuke submitted to support later versions. So we can carry forward.

@lixuna

**Comment by lixuna** [comment]
@agentpoyo @HashNuke As of today, the Latest release is [1.11.2](https://crystal-lang.org/2024/01/18/1.11.2-released/)

Can this maintenance item be prioritized for Q1?

**Comment by agentpoyo** [comment]
> @agentpoyo @HashNuke As of today, the Latest release is [1.11.2](https://crystal-lang.org/2024/01/18/1.11.2-released/)
> 
> Can this maintenance item be prioritized for Q1?

Yeah, I think we can work on this. @HashNuke sound good?

**Comment by lixuna** [comment]
@agentpoyo @HashNuke Please share the status of this issue. Thank you :)

**Comment by agentpoyo** [comment]
> @agentpoyo @HashNuke Please share the status of this issue. Thank you :)

I can start the testing of latest crystal and @HashNuke can probably jump in as well.
